internal/controller/telegram/telegram.go

```diff
@@ -0,0 +1,189 @@
+package telegram
+
+import (
+	"context"
+	"fmt"
+	"github.com/evrone/go-clean-template/internal/entity"
+	"github.com/evrone/go-clean-template/internal/usecase"
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api/v5"
+	"github.com/nfnt/resize"
+	_ "github.com/nfnt/resize"
+	"image"
+	"image/jpeg"
+	_ "image/jpeg"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+type Telegram struct {
+	BotAPI    *tgbotapi.BotAPI
+	botToken  string
+	useCase   usecase.Recognition
+	newAnswer chan bool
+	classes   *entity.Classes
+}
+
+const layout = "2006_01_02"
+const filePath = "pkg/file_storage/images/"
+
+func New(token string, useCase usecase.Recognition, classes *entity.Classes, newAnswer chan bool, opts ...Option) (*Telegram, error) {
+
+	botAPI, err := tgbotapi.NewBotAPI(token)
+
+	if err != nil {
+		return nil, err
+	}
+
+	tg := &Telegram{
+		botToken:  token,
+		BotAPI:    botAPI,
+		useCase:   useCase,
+		newAnswer: newAnswer,
+		classes:   classes,
+	}
+	// Custom options
+	for _, opt := range opts {
+		opt(tg)
+	}
+	log.Printf("Authorized on account %s", tg.BotAPI.Self.UserName)
+
+	return tg, err
+}
+
+func (tg *Telegram) Start() {
+	u := tgbotapi.NewUpdate(0)
+	u.Timeout = 60
+	updates := tg.BotAPI.GetUpdatesChan(u)
+	ctx := context.Background()
+	go func() {
+		for update := range updates {
+
+			if update.Message != nil { // If we got a message
+				// Если сообщение содержит изображение
+				if update.Message.Photo != nil {
+					// Получаем информацию об изображении
+					fileID := (update.Message.Photo)[len(update.Message.Photo)-1].FileID
+					file, err := tg.BotAPI.GetFile(tgbotapi.FileConfig{FileID: fileID})
+					if err != nil {
+						log.Println(err)
+						continue
+					}
+
+					// Загружаем изображение
+					imagePathName, err := downloadFile(file.Link(tg.BotAPI.Token), strconv.Itoa(update.Message.MessageID))
+					if err != nil {
+						log.Println(err)
+						continue
+					}
+
+					// Add record to pg for worker
+					newReq := entity.Request{
+						ChatID:        update.Message.Chat.ID,
+						ImagePathName: imagePathName,
+					}
+					err = tg.useCase.AddRequest(ctx, newReq)
+					if err != nil {
+						log.Println(err)
+					}
+					// Отправляем сообщение об успешной загрузке изображения
+					msg := tgbotapi.NewMessage(update.Message.Chat.ID, "Изображение успешно загружено на сервер....")
+					_, err = tg.BotAPI.Send(msg)
+					if err != nil {
+						log.Println(err)
+					}
+
+				}
+			}
+		}
+	}()
+	for {
+		if <-tg.newAnswer {
```

Comment:
```suggestion
	for _ = range <-tg.newAnswer {
```

для чтения из канала достаточно просто читать из него, дополнительные проверки не нужны